src/classes/CRLP_ApiService.cls

```diff
@@ -194,8 +218,274 @@ public without sharing class CRLP_ApiService {
+    */
+    private RollupMetadataHandler getRollupHandler(Set<String> changedRollupTypes) {
+        RollupMetadataHandler callback = null;
+
+         if (!changedRollupTypes.isEmpty()) {
+            callback = new RollupMetadataHandler();
+            callback.params = new Map<String, Set<String>> { 
+                PARAM_ROLLUP_TYPES => changedRollupTypes
+            };
+        }
+
+        return callback;
+    }
+
+
+    /*
+    * @description handle changes to customizable rollups and send back a callback handler if 
+    * there is processing that needs to be executed after deloyment
+    * @param rollups is a List of Rollups
+    * @return a MetadataHandler that can be used for deployment callbacks or null
+    */
+    public RollupMetadataHandler sendChangeEvent(List<CRLP_RollupCMT.Rollup> rollups) {
+        if (!isRollupStateEnabled()) {
+            return null;
+        }
+
+        Set<String> changedRollupTypes = new Set<String>();
+
+        for (CRLP_RollupCMT.Rollup cmtRollup : rollups) {
+            RollupUtil util = new RollupUtil(cmtRollup);
+            if (util.hasChanged()) {
+                changedRollupTypes.addAll(util.getRollupTypes());
+            }
+        }
+
+       return getRollupHandler(changedRollupTypes);
+    }
+
+
+    /*
+    * @description handle changes to customizable filter groups and send back a callback handler if
+    * there is processing that needs to be executed after deployment
+    * @param filterGroups is a list of filtergroups
+    * @return a MetadataHandler that can be used for deployment callbacks or null
+    */
+    public RollupMetadataHandler sendChangeEvent(List<CRLP_RollupCMT.FilterGroup> filterGroups) {
+        if (!isRollupStateEnabled()) {
+            return null;
+        }
+
+        Set<String> changedRollupTypes = new Set<String>();
+
+        FilterGroupUtil util = new FilterGroupUtil();
+        for (CRLP_RollupCMT.FilterGroup cmtFilterGroup : filterGroups) {
+            if (util.hasChanged(cmtFilterGroup)) {
+                changedRollupTypes.addAll(util.getRollupTypes(cmtFilterGroup.recordId));
+            }
+        }
+
+        return getRollupHandler(changedRollupTypes);
+    }
+
+
+
+    /*******************************************************************************************************
+    * @description Utility class for a CMT rollup
+    */
+    public class RollupUtil {
+        CRLP_RollupCMT.Rollup cmtRollup;
+        private CRLP_RollupProcessingOptions.RollupType oldRollupType = null;
+
+        private CRLP_RollupProcessingOptions.RollupType rollupType {
+            get {
+                if (rollupType == null) {
+                    rollupType = cmtRollup.getRollupType();
+                }
+                return rollupType;
+            }
+            set;
+        }
+
+        private Boolean isNew {
+            get {
+                if (isNew == null) {
+                    isNew = String.isBlank(cmtRollup.recordName);
```

Comment:
I'm pretty sure that `recordName` gets assigned immediately in the constructor for `CRLP_RollupCMT.Rollup`, so it should never be blank. What you could probably do is add a Boolean to the Rollup, FilterGroup, FilterRule inner classes for `isNew` that gets set to true in the constructor that accepts a String label value.

Reply:
Changes for the rollup `isNew` have been reverted back. Creating new or cloning existing rollup work only when used from `RollupUtil` as is. Please let me know if there are questions.

Reply:
@RadSFDeveloper Not sure I follow this. In the `CRLP_RollupCMT.Rollup` inner class, the constructor - when creating a new rollup - is the following (line 283):

```
        public Rollup(String label) {
            //check to see if UDR; these need a unique name
            Boolean isUDR = (label.left(3) == 'UDR');
            this.label = label.replace('Number', 'Num').left(40).trim();
            this.recordName = generateRecordName(this.label, isUDR);
            this.useFiscalYear = false;
            this.isActive = true;
        }
```

Based on the above, I don't believe the `recordName` property would ever be empty. If true, then `isNew` could never return true.